src/Orleans.Core.Abstractions/Orleans.Core.Abstractions.csproj

```diff
@@ -11,5 +11,9 @@
     <RootNamespace>Orleans</RootNamespace>
     <TargetFramework>netstandard1.3</TargetFramework>
   </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="1.1.1" />
+  </ItemGroup>
```

Comment:
Why move `ISiloBuilder` to abstractions (which introduced this dependency too)?
At least in our current design (https://github.com/dotnet/orleans/issues/3353), the idea is not to have the building and configuration be part of the abstractions project. Otherwise this would make as rev this abstractions package project way more often.
We are potentially considering having a separate abstraction for hosting, but until then, everything related to hosting should go into Core.

Reply:
ISiloBuilder was moved so I could add transaction feature setup that was compatible with the builder pattern.
The general problem:  How should extension libraries register services with the silo builder?
Pls see SiloBuilderExtensions  and DevelopmentSiloBuilderExtensions.
I can remove these extension an provide only configuration services actions if that is preferred, but that seems a bit clumsy.

Reply:
Well, the thing is that this project can reference Core (in addition to Core.Abstractions) and still define the extension methods (which you shouldn't remove).
At some point we can have some Microsoft.Orleans.Hosting.Abstractions (although I wouldn't go that way yet until we start using the generic hosting abstractions from the aspnet team). We could also (instead or in addition to) create a Microsoft.Orleans.Transactions.Abstractions so that grain developers would reference just that, instead of Transactions, which now brings a dependency to Core. Nevertheless, that's just going to add more issues than benefits for end users, at least for now until we know the abstractions are absolutely stable and we can guarantee no breaking changes.

Reply:
Will attempt to use runtime dependency.  Should be ok I think.